### ISSUE TITLE
GCD throwing false warning messages on exporting cross section database

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/gcd#313]
## The Problem
When you attempt to export the database, it says its not working:
![error](https://user-images.githubusercontent.com/9723905/44591088-0403b280-a77a-11e8-9099-834e003a7029.png)

Followed by more worrying messages:
![error2](https://user-images.githubusercontent.com/9723905/44591094-06660c80-a77a-11e8-8318-7ece106661cf.png)

Even though it actually creates the thing on disc. Watch the video to see what happens here and subsequently in Cross Section Explorer:

https://youtu.be/Tpz2xe0zNJ0

## Anything else?
See related [issue 9 in XS Explorer](https://github.com/NorthArrowResearch/xsviewer-issues/issues/9)

**Comment by philipbaileynar** [comment]
Can I get a zip of the project please?

**Comment by philipbaileynar** [comment]
@joewheaton I am unable to produce.

1. Can you please confirm your versions:
    * GCD version 7.1.1
    * Cross Section Viewer 2.11
1. I used the [Ngaruroro dataset directly from the Aberdeen workshop](http://gcd.riverscapes.xyz/Help/Workshops/workshop-schedules/2018/Aberdeen/excercises.html). I literally opened the project and immediately tried to export to the Cross Section Viewer software.
1. It occurred to me that there are **no linear extractions** in this project and so there's really nothing to export! But I still got an empty database and the software opened.

![2018-08-24_160826](https://user-images.githubusercontent.com/8645379/44611737-f2410000-a7b7-11e8-8d5c-ce841c6d1695.png)

**Comment by jb10016** [comment]
@philipbaileynar @joewheaton Only just seen this now guys ... this Philip - Joe has rather neatly picked up the second problem I referred to in #318 .

**Comment by nicolewheelerhydrobiology** [comment]
I've had the exact same problem described above, when trying to export to the cross section viewer both from GCD within ArcMap and from the GCD stand alone. 

I found sometimes opening a new ArcMap document and starting a new GCD project and starting from scratch appeared to fix this problem but not always. 

Have you guys come up with a work around for this issue? I am currently stuck....

**Comment by philipbaileynar** [comment]
@nicolewheelerhydrobiology I can help you with this issue but can I request some more information please?

1. GCD Version?
1. ArcGIS Version?
1. Can you zip up your GCD project and send it to me please? It might be too big for GitHub issues.
1. Step by step instructions to reproduce the problem. Starting with "Open GCD standalone..."
1. Any screenshots of your particular error messages? If there's a "details button" then click it and cut and paste the contents of the text into this issue.

**Comment by nicolewheelerhydrobiology** [comment]
1. GCD Version 7.4.1. (Latest release, downloaded the other day) and Cross Section Viewer 2.0 Beta Download

2. ArcMap 10.6 

3. [BoundaryCreekDifferencingXsections0311new2.zip](https://github.com/Riverscapes/gcd/files/3454301/BoundaryCreekDifferencingXsections0311new2.zip)

4. Steps:
- open GCD standalone (or ArcMap)
- add 2003 and 2011 clipped DEMs
- add existing transect profile routes: xsections2.shp - set Station values to 'Distance' and Label field 'River Stat'; extraxsectionexport.shp - set Station values to 'Distance' and Label field to 'Label'
- Calculate linear extraction from profile route for each of the Dems, for each of the 2 cross section files
- Export cross sections to cross section viewer

5. The following error messages appear:

![image](https://user-images.githubusercontent.com/53249059/62256247-e63bd480-b443-11e9-928a-869d6e2d113c.png)

![image](https://user-images.githubusercontent.com/53249059/62256281-fb186800-b443-11e9-9846-21d72025f7b7.png)

![image](https://user-images.githubusercontent.com/53249059/62256314-17b4a000-b444-11e9-863a-981bcbcac4e3.png)

Details:

See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.ArgumentNullException: Value cannot be null.
Parameter name: source
   at System.Linq.Enumerable.Where[TSource](IEnumerable`1 source, Func`2 predicate)
   at CrossSectionViewer.Tools.frmSurfaceDelta.LoadDataControls(Object sender, EventArgs e)
   at CrossSectionViewer.Tools.frmSurfaceDelta.frmSurfaceDelta_Load(Object sender, EventArgs e)
   at System.Windows.Forms.Form.OnLoad(EventArgs e)
   at System.Windows.Forms.Form.OnCreateControl()
   at System.Windows.Forms.Control.CreateControl(Boolean fIgnoreVisible)
   at System.Windows.Forms.Control.CreateControl()
   at System.Windows.Forms.Control.WmShowWindow(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Form.WmShowWindow(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3815.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3801.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
CrossSectionViewer
    Assembly Version: 2.0.13.0
    Win32 Version: 2.0.13.0
    CodeBase: file:///C:/Users/GIS/AppData/Local/Apps/2.0/8W5H79JY.1BT/2NLGJ7N7.9L0/cros..tion_d2a9a24e25080d83_0001.0000_f70b316455167fb1/CrossSectionViewer.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3815.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3815.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
xsUI
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/Users/GIS/AppData/Local/Apps/2.0/8W5H79JY.1BT/2NLGJ7N7.9L0/cros..tion_d2a9a24e25080d83_0001.0000_f70b316455167fb1/xsUI.DLL
----------------------------------------
xslib
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/Users/GIS/AppData/Local/Apps/2.0/8W5H79JY.1BT/2NLGJ7N7.9L0/cros..tion_d2a9a24e25080d83_0001.0000_f70b316455167fb1/xslib.DLL
----------------------------------------
System.Data.SQLite
    Assembly Version: 1.0.108.0
    Win32 Version: 1.0.108.0
    CodeBase: file:///C:/Users/GIS/AppData/Local/Apps/2.0/8W5H79JY.1BT/2NLGJ7N7.9L0/cros..tion_d2a9a24e25080d83_0001.0000_f70b316455167fb1/System.Data.SQLite.DLL
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.Transactions
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Transactions/v4.0_4.0.0.0__b77a5c561934e089/System.Transactions.dll
----------------------------------------
Microsoft.VisualBasic
    Assembly Version: 10.0.0.0
    Win32 Version: 14.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualBasic/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualBasic.dll
----------------------------------------
UnitsNet
    Assembly Version: 3.77.0.0
    Win32 Version: 3.77.0.0
    CodeBase: file:///C:/Users/GIS/AppData/Local/Apps/2.0/8W5H79JY.1BT/2NLGJ7N7.9L0/cros..tion_d2a9a24e25080d83_0001.0000_f70b316455167fb1/UnitsNet.DLL
----------------------------------------
System.EnterpriseServices
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.EnterpriseServices/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.EnterpriseServices.dll
----------------------------------------
naru
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///C:/Users/GIS/AppData/Local/Apps/2.0/8W5H79JY.1BT/2NLGJ7N7.9L0/cros..tion_d2a9a24e25080d83_0001.0000_f70b316455167fb1/naru.DLL
----------------------------------------
System.Windows.Forms.DataVisualization
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3815.0
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.DataVisualization/v4.0_4.0.0.0__31bf3856ad364e35/System.Windows.Forms.DataVisualization.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.

**Comment by philipbaileynar** [comment]
Thank you @nicolewheelerhydrobiology that's a fantastic goldmine of information. I have downloaded the project and the reviewed the DEMs and profile routes. Everything looks correct in that regard.

Can you humour me and try the following:

1. Close GCD Standalone
1. Close ArcMap
1. Close Cross Section Viewer
1. Using Windows Explorer please move the GCD to a folder location that doesn't contain any spaces. Top level of the C:\ drive is best, but any folder path without a space should suffice. Basically move the entire folder that contains the *.gcd project file to a path that does not contain spaces.
1. Open either version of the GCD and try the export to Cross Section Viewer again.

Please report back if this solves the issue or not.

ps. Lots of people are using this feature successfully, so I know that we can get you up and running.

**Comment by nicolewheelerhydrobiology** [comment]
Hi @philipbaileynar 

Thanks for getting back to me. I tried what you suggested, and got similar error messages:

![image](https://user-images.githubusercontent.com/53249059/62333168-bbb25000-b505-11e9-8d50-9745f7b1f5ef.png)
![image](https://user-images.githubusercontent.com/53249059/62333203-e00e2c80-b505-11e9-9ccf-d573e302d7f7.png)

However, I will try starting a completely new project from scratch at the top level of the folder without spaces and see how I get on. I'll give you an update on how that goes.

I've also managed to use this feature successfully in the past, by just starting again and for some reason it working. So hopefully I can this up and running.
